lib/internal/modules/esm/translators.js

```diff
@@ -276,26 +316,37 @@ translators.set('commonjs', function commonjsStrategy(url, source, isMain) {
   return createCJSModuleWrap(url, source, isMain, cjsLoader);
 });
 
+/**
+ * Get or create an entry in the CJS module cache for the given filename.
+ * @param {string} filename CJS module filename
+ * @returns {CJSModule} the cached CJS module entry
+ */
+function cjsEmplaceModuleCacheEntry(filename, exportNames) {
+  // TODO: Do we want to keep hitting the user mutable CJS loader here?
+  let module = CJSModule._cache[filename];
```

Comment:
Since we are already moving code here, would you mind changing them to `mod` or something that isn't `module`? This shadowing of the top-level `module` variable in the inner functions always feel very bug-prone to me, and I have found & removed some dead code in the loader before that was referencing the top-level `module` variable even though it was meant to reference a shadowing `module` variable that was removed, but the linter would not complain about referencing undefined variables due to the shadowing. Doesn't have to be done in this PR, but I just cannot ignore the shadowing ;)

Reply:
thanks, updated!